HIP/hip-1056.md

```diff
@@ -0,0 +1,1822 @@
+    }
+}
+```
+
+```protobuf
+/**
+ * Removal of an item from a `Queue` state.<br/>
+ */
+message QueuePopChange {
+}
+```
+
+> Note: The block stream design ensures consistency of singleton and queue states at round boundaries so as to reduce the
+block stream size. That is singleton and queue state changes are written to the block stream at the end of a round.
+The platform computes the state hash at round boundaries, and the consensus node tracks these hashes. When a block ends
+the node uses the hash from the last non-empty round as the `start_of_block_state_root_hash` in the BlockProof.
+With that in mind, while intermediate states may temporarily diverge during block stream replay, the final state after
+applying all state changes in a round will match the live network's state at that same round boundary.
+
+#### FilteredBlockItem
+
+The block stream will support the ability to omit `BlockItem`s from the stream while retaining full verification of the
+block proof. This is possible because all items are double hashed. Their hash is hashed into the parent node not the raw
+contents of the item. The block proof is based on the signing of the root hash of a block merkle tree, therefore any
+item or subtree in the proof can be replaced by a `FilteredItemHash`. Notably, the replacement of an item with a hash
+will not impact the ability for downstream services to validate block proofs.
+
+
+```protobuf
+message FilteredItemHash {
+    /**
+    * A hash of an item filtered from the stream.
+    *
+    * The hash algorithm used MUST match the hash algorithm specified in
+    * the block header for the containing block.
+    */
+    bytes item_hash = 1;
+    
+    /**
+    * A binary tree path to the tree node that is missing and replaced by the hash above.
+    */
+    uint64 filtered_path = 3;
+}
+```
+
+> 💡 *Sidebar: Subsetting and Errata for Block Stream.*
+Block stream supports subsetting the block stream through the use of a `FilteredItemHash` to
+replace any item that has been removed. The removed item is thus replaced by its hash and the path that was filtered
+(some removals may require simultaneous removal of a small subtree), which preserves the ability to verify the
+integrity of the block stream and block proof.  Errata are handled similarly for removed items. Adding items *after the
+fact* or replacing an erroneous item requires potentially removing a prior item, and adding the correction (errata) as a
+new item in the block stream at the point of the “errata” transaction, which is then part of the block proof for the
+block where the “errata” was processed.
+
+#### Block Proof
+
+The block proof carries the trust of the consensus network to the contents of the block. A block can conceptually be
+represented as a binary merkle tree with 4 sub-trees at depth 2 which create the block proof and tree when hashed to
+obtain a merkle root.
+
+![Root Hashes](../assets/hip-1056/root_hashes.svg)
```

Comment:
```suggestion
![Root Hashes](../assets/hip-1056/root-hashes.svg)
```
The SVG file has two `Root of Input Merkle Tree`, and so missing `Root of Output Merkle Tree`